src/meta-srv/src/procedure/state_store.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use api::v1::meta::{BatchDeleteRequest, PutRequest, RangeRequest};
+use async_stream::try_stream;
+use async_trait::async_trait;
+use common_procedure::error::{
+    CorruptedDataSnafu, DeleteStatesSnafu, ListStateSnafu, PutStateSnafu,
+};
+use common_procedure::store::state_store::{KeyValueStream, StateStore};
+use common_procedure::Result;
+use snafu::ResultExt;
+
+use crate::service::store::kv::KvStoreRef;
+use crate::util;
+
+const PROCEDURE_PREFIX: &str = "/__procedure__/";
+
+fn with_prefix(key: &str) -> String {
+    format!("{PROCEDURE_PREFIX}{key}")
+}
+
+fn strip_prefix(key: &str) -> String {
+    key.trim_start_matches(PROCEDURE_PREFIX).to_string()
+}
+
+pub(crate) struct MetaStateStore {
+    kv_store: KvStoreRef,
+    max_size_per_range: i64,
+}
+
+impl MetaStateStore {
+    pub(crate) fn new(kv_store: KvStoreRef) -> Self {
+        Self {
+            kv_store,
+            max_size_per_range: -1,
+        }
+    }
+}
+
+#[async_trait]
+impl StateStore for MetaStateStore {
+    async fn put(&self, key: &str, value: Vec<u8>) -> Result<()> {
+        let _ = self
```

Comment:
nitpick: Do we need `let _`?
```suggestion
        self
```

Reply:
`put` returns a value we need to consume to make clippy happy